crates/amaru-ledger/src/rules/transaction/output_size.rs

```diff
@@ -0,0 +1,45 @@
+use amaru_kernel::{
+    cbor, output_lovelace, protocol_parameters::ProtocolParameters, TransactionBody,
+    TransactionOutput,
+};
+
+use crate::rules::TransactionRuleViolation;
+
+pub struct OutputTooSmall {
+    pub outputs_too_small: Vec<TransactionOutput>,
+}
+
+impl Into<TransactionRuleViolation> for OutputTooSmall {
+    fn into(self) -> TransactionRuleViolation {
+        TransactionRuleViolation::OutputTooSmall(self)
+    }
+}
+
+// clean up clones, replace with lifetimes
+#[allow(clippy::panic)]
+pub fn validate_output_size(
+    transaction: &TransactionBody,
+    protocol_parameters: &ProtocolParameters,
+) -> Result<(), OutputTooSmall> {
+    let coins_per_utxo_byte = protocol_parameters.coins_per_utxo_byte;
+    let outputs_too_small = transaction
+        .outputs
+        .clone()
+        .into_iter()
+        .filter(|output| {
+            let mut bytes = Vec::new();
+            cbor::encode(output, &mut bytes)
+                .unwrap_or_else(|_| panic!("Failed to serialize output"));
```

Comment:
It would make sense to handle this error via proper error handling now. It would force us to figure out what's the proper way to handle this case and design a proper API. 
We probably don't want to crash the node here.

i understand it's a PITA, but it's one we want to solve sooner rather than later IMO